lib/net.js

```diff
@@ -740,38 +745,15 @@ Socket.prototype._writeGeneric = function(writev, data, encoding, cb) {
     return false;
   }
 
-  var req = new WriteWrap();
-  req.handle = this._handle;
-  req.oncomplete = afterWrite;
-  req.async = false;
-  var err;
-
-  if (writev) {
-    var allBuffers = data.allBuffers;
-    var chunks;
-    var i;
-    if (allBuffers) {
-      chunks = data;
-      for (i = 0; i < data.length; i++)
-        data[i] = data[i].chunk;
-    } else {
-      chunks = new Array(data.length << 1);
-      for (i = 0; i < data.length; i++) {
-        var entry = data[i];
-        chunks[i * 2] = entry.chunk;
-        chunks[i * 2 + 1] = entry.encoding;
-      }
-    }
-    err = this._handle.writev(req, chunks, allBuffers);
-
-    // Retain chunks
-    if (err === 0) req._chunks = chunks;
-  } else {
-    err = createWriteReq(req, this._handle, data, encoding);
-  }
+  let ret;
+  const req = createWriteWrap(this._handle, afterWrite);
+  if (writev)
+    ret = writevGeneric(this, req, data, cb);
+  else
+    ret = writeGeneric(this, req, data, encoding, cb);
 
-  if (err)
-    return this.destroy(errnoException(err, 'write', req.error), cb);
+  // this only runs when writeGeneric returns error.
```

Comment:
Nit: the comment is a bit misleading in a couple of ways:

1. It can also run when `writevGeneric` fails.
2. Both `writeGeneric` and `writeGeneric` return `self` when an error occurs instead of the actual error.

I'd use something like "Bail out if `handle.write*` returned an error".